### PR TITLE
Add ability to delete tso datasets for zOS guests

### PR DESCRIPTION
**Pull request by jdekonin**
* [skip ci]

Signed-off-by: Joe deKoning <joe_dekoning@ca.ibm.com>

### CODE REVIEW

**Comment by jdekonin** [comment]
@vsebe @AdamBrousseau @pshipton for review

**Review by AdamBrousseau — CHANGES_REQUESTED** [review]
Indenting uses tabs instead of spaces
image

**Comment by jdekonin** [comment]
> Indenting uses tabs instead of spaces

Done.

File: buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy [hunk 1]
```diff
@@ -179,6 +179,14 @@ timeout(time: TIMEOUT_TIME.toInteger(), unit: TIMEOUT_UNITS) {
                                         }
                                     }
 
+                                    // Cleanup zOS datasets 
+                                    if (nodeLabels.contains('sw.os.zos')) {
+                                        listcat = sh(script: "tso listcat | grep '${env.USER}' | cut -d. -f 2-", returnStdout: true).trim()
```
Comment (pshipton) [inline buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy:184]: Are you sure this works? It seems like it should but when I try `tso listcat | grep JENKINS` manually, it doesn't find anything although there is a dataset there.

Comment (pshipton) [reply buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy:184]: Trying something else `tso listcat | tee out` it outputs garbage, making me thing there is a ascii/ebcdic issue.

`tso listcat > out && grep JENKINS out` works.

Comment (pshipton) [reply buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy:184]: I was using fyrec50g

Comment (pshipton) [reply buildenv/jenkins/jobs/infrastructure/Cleanup-Nodes.groovy:184]: Actually `sh -c "tso listcat | grep JENKINS"` works, so I guess this will too.

**Review by AdamBrousseau — APPROVED** [review]
lftm assuming the commands work as expected.

**Review by pshipton — APPROVED** [review]

**Pull request merged**